server/src/main/java/keywhiz/service/crypto/CryptoModule.java

```diff
@@ -54,7 +55,9 @@ public CryptoModule(String derivationProviderClass, KeyStoreConfig keyStoreConfi
     this.keyStoreConfig = keyStoreConfig;
   }
 
-  @Override protected void configure() {}
+  @Override protected void configure() {
+    bind(SecureRandom.class).toProvider(SecureRandom::new);
+  }
```

Comment:
Previously, Guice would silently create a new SecureRandom whenever needed because it has a no-argument constructor. While helpful, this can mask silent bugs, one of which I encountered while putting this PR together, so it's not just an academic concern. Now we have to explicitly tell Guice that we want to create new SecureRandoms via the no-arg constructor.